documentation/architecture/adr-009-translation-system.md

```diff
@@ -0,0 +1,110 @@
+GitHub. To detect new UI strings a GitHub Actions workflow scans the codebase
+for new strings and notifies POEditor. Here they can be translated by
+non-technical users. POEditor supports committing translations back to GitHub
+where they can be consumed by DPL CMS instances.
+
+## Consequences
+
+This approach has a number of benefits apart from addressing the original
+issue:
+
+- POEditor is a specialized tool to manage translations. It supports features
+  such as translation memory, glossaries and machine translation.
+- POEditor is web-based. Translators avoid having to find and install a suitable
+  tool to edit `.po` files.
+- POEditor is software-as-a-service. We do not need to maintain the translation
+  interface ourselves.
+- POEditor is free for open source projects. This means that we can use it
+  without having to pay for a license.
+- Code scanning means that new UI strings are automatically detected and
+  available for translation. We do not have to manually synchronize translation
+  files or ensure that UI strings are rendered by the system before they can be
+  translated. This can be complex when working with special cases, error
+  messages etc.
+- Translations are stored in version control. Managing state is complex and this
+  means that we have easy visibility into changes.
+- Translations are stored on GitHub. We can move away from POEditor at any time
+  and still have access to all translations.
+- We reuse existing systems instead of building our own.
+
+A consequence of this approach is that developers have to write code that
+supports scanning. This is partly supported by the Drupal Code Standards. To
+support contexts developers also have to include these as a part of the `t()`
+function call e.g.
+
+```php
+// Good
+$this->t('A string to be translated', [], ['context' => 'The context']);
+$this->t('Another string', [], ['context' => 'The context']);
+// Bad
+$c = ['context' => 'The context']
+$this->t('A string to be translated', [], $c);
+$this->t('Another string', [], $c);
+```
+
+We could consider writing a custom sniff or PHPStan rule to enforce this
+
+### Potion
+
+For covering the functionality of scanning the code we had two potential
+projects that could solve the case:
+
+- [Potion](https://www.drupal.org/project/potion)
+- [Potx](https://www.drupal.org/project/potx)
+
+Both projects can scan the codebase and generate a `.po` or `.pot` file with the
+translation strings and context.
+
+At first it made most sense to go for Potx since it is used by
+[localize.drupal.org](https://localize.drupal.org) and it has a long history.
+But Potx is extracting strings to a `.pot` file without having the possibility
+of filling in the existing translations. So we ended up using Potion which can
+fill in the existing strings.
+
+A flip side using Potion is that it is not being maintained anymore. But it
+seems quite stable and a lot of work has been put into it. We could consider to
+back it ourselves.
```

Comment:
I think DDF need to take over that module... we should expect many new Drupal and PHP versions in the life time of this project. So I think you need to find a solution for this now with DDF...